doc/examples/uuid.rst

```diff
@@ -0,0 +1,290 @@
+specified UUID representation to insert the example UUID
+``00112233-4455-6677-8899-aabbccddeeff``into MongoDB. If we try to read this
+value using PyMongo with no UUID representation specified, we end up with an
+entirely different UUID::
+
+  UUID('77665544-3322-1100-ffee-ddccbbaa9988')
+
+However, if we explicitly set the representation to
+:data:`~bson.binary.UuidRepresentation.JAVA_LEGACY`, we get the correct result::
+
+  UUID('00112233-4455-6677-8899-aabbccddeeff')
+
+PyMongo uses the specified UUID representation to reorder the BSON bytes and
+load them correctly. ``JAVA_LEGACY`` encodes native :class:`uuid.UUID`s to
+:class:`~bson.binary.Binary` subtype 3 objects, while performing the same
+byte-reordering as the legacy Java driver's UUID to BSON encoder.
+
+The ``CSHARP_LEGACY`` UUID Representation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. attention:: This uuid representation should be used when reading UUIDs
+   written to MongoDB by the C# driver without an explicitly configured UUID
+   representation.
+
+The :data:`~bson.binary.UuidRepresentation.CSHARP_LEGACY` representation
+corresponds to the legacy representation of UUIDs used by the MongoDB Java
+Driver.
+
+.. note:: The ``CSHARP_LEGACY`` representation reverses the order of bytes 0-3,
+   bytes 4-5, and bytes 6-7.
+
+As an example, consider the same UUID described in :ref:`example-legacy-uuid`.
+Let us assume that an application used the C# driver without an explicitly
+specified UUID representation to insert the example UUID
+``00112233-4455-6677-8899-aabbccddeeff``into MongoDB. If we try to read this
+value using PyMongo with no UUID representation specified, we end up with an
+entirely different UUID::
+
+  UUID('33221100-5544-7766-8899-aabbccddeeff')
+
+However, if we explicitly set the representation to
+:data:`~bson.binary.UuidRepresentation.CSHARP_LEGACY`, we get the correct result::
+
+  UUID('00112233-4455-6677-8899-aabbccddeeff')
+
+PyMongo uses the specified UUID representation to reorder the BSON bytes and
+load them correctly. ``CSHARP_LEGACY`` encodes native :class:`uuid.UUID`s to
+:class:`~bson.binary.Binary` subtype 3 objects, while performing the same
+byte-reordering as the legacy C# driver's UUID to BSON encoder.
+
+The ``STANDARD`` UUID Representation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. attention:: This uuid representation should be used by new applications
+   that have never stored UUIDs in MongoDB.
+
+The :data:`~bson.binary.UuidRepresentation.STANDARD` representation
+enables cross-language compatibility by ensuring the same byte-ordering
+when encoding UUIDs from all drivers. UUIDs written by a driver with this
+representation configured can be read by every other driver correctly provided
+it is configured with the ``STANDARD`` representation.
+
+``STANDARD`` encodes native :class:`uuid.UUID`s to
+:class:`~bson.binary.Binary` subtype 4 objects.
+
+
+The ``UNSPECIFIED`` UUID Representation
```

Comment:
Can you add a code example of what happens when you read a document containing a UUID with UNSPECIFIED and another example of what happens when you attempt (and fail) to write a document with a UUID? I think it would be nice to include the full error traceback here.

Reply:
You should also mention, if you haven't already - I might have missed it, that UNSPECIFIED will become the default in PyMongo 4.0.